### DISCUSSION TITLE
Plans to support persistent connection?

### DISCUSSION THREAD

**Discussion by mcrumiller** [sfu-db/connector-x#317] [Ideas]
Currently, to connect to a database, a connection string must be supplied, which creates a brand new connection each sql query. This makes it impossible to, for example, create temporary tables for use in future queries.

Are there any plans to support connections such as those provided by [SQLAlchemy](https://www.sqlalchemy.org/)?

**Comment by wangxiaoying** [comment]
Hi @mcrumiller , thanks for the suggestion! Indeed without persistent connection, there will be an issue handling temporary tables. We will consider this in our future plan.
